cmake/GenerateVersion.cmake

```diff
@@ -0,0 +1,47 @@
+# Handle generating version from Git.
+set(use_git_version 0)
+if(NVIM_VERSION_MEDIUM)
+  message(STATUS "USING NVIM_VERSION_MEDIUM: ${NVIM_VERSION_MEDIUM}")
+  return()
+endif()
+
+find_program(GIT_EXECUTABLE git)
+if(NOT GIT_EXECUTABLE)
+  message(AUTHOR_WARNING "Skipping version-string generation (cannot find git)")
+  return()
+endif()
+
+execute_process(
+  COMMAND git describe --first-parent --tags --always --dirty
+  OUTPUT_VARIABLE GIT_TAG
+  ERROR_VARIABLE ERR
+  RESULT_VARIABLE RES
+)
+
+if("${RES}" EQUAL 1)
```

Comment:
@kylo252 Should this be checking `!= 0` instead of `== 1`?  `git describe` outside of a repo has an exit code of 128.